### PR TITLE
[#119570591] Pre-populate Sanger Customer Sample IDs

### PR DESCRIPTION
**Pull request by jhanggi**
This extends from #625, which is where most of the commits come from. Only the last two are specific to this PR.

### CODE REVIEW

**Comment by jhanggi** [comment]
Rebased off of master now that #625 is in.

File: vendor/engines/sanger_sequencing/app/models/sanger_sequencing/sample.rb [hunk 1]
```diff
@@ -4,6 +4,20 @@ class Sample < ActiveRecord::Base
...
+
+    def customer_sample_id
+      self[:customer_sample_id] ||= default_customer_sample_id
+    end
+
+    private
+
+    def default_customer_sample_id
+      # last four digits of the id as a zero-padded string
+      id.to_s.last(4).rjust(4, "0")
```
Comment (drench) [inline vendor/engines/sanger_sequencing/app/models/sanger_sequencing/sample.rb]: I don't know if it's much more clear but I would have gone with `format("%04d", id).last(4)`, presuming `id` won't never be `nil` here.

File: vendor/engines/sanger_sequencing/spec/features/purchasing_spec.rb [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
+      click_link "Complete Online Order Form"
+    end
+
+    it "sets up the right number of text boxes" do
+      expect(page).to have_css(customer_id_selector, count: 5)
+    end
+
+    it "has prefilled values in the text boxes with unique four digit numbers" do
+      values = page.all(customer_id_selector).map(&:value)
+      expect(values).to all(match(/\d{4}/))
```
Comment (drench) [inline vendor/engines/sanger_sequencing/spec/features/purchasing_spec.rb]: `match(/\A\d{4}\z/)` to be sure they're all four, and only four digits?

**Comment by drench** [comment]
A couple comments but nothing big.

File: vendor/engines/sanger_sequencing/app/models/sanger_sequencing/sample.rb (hunk 1 above)
Comment (jhanggi) [inline vendor/engines/sanger_sequencing/app/models/sanger_sequencing/sample.rb]: Yeah, I like that. I always forget about `format`. I'm fine erroring if `id` is nil, as that would represent an actual bug.

**Comment by jhanggi** [comment]
Feedback addressed.

**Comment by drench** [comment]
:+1:

**Pull request merged**